### PR TITLE
feat: PermissionsWorkflow

### PR DESCRIPTION
**Pull request by drymarau**
- refactor: WelcomeWorkflow -> DisplayNameWorkflow
- feat: PermissionsWorkflow

### CODE REVIEW

File: sample/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 <resources>
     <string name="app_name">"Video SDK"</string>
+    <string name="app_name_full">"Pexip Video SDK"</string>
     <!-- CONFERENCE -->
     <string name="conference_notification_channel_id" translatable="false">conference</string>
     <string name="conference_notification_channel_name">Conference</string>
     <string name="conference_notification_content_title">Conference</string>
+    <!-- PERMISSIONS -->
+    <string name="permissions_title">Grant permissions</string>
+    <string name="permissions_description">Granting permission in the next screen will make sure other participants are able to see and hear you if you’ve enabled your camera and microphone</string>
```
Comment (tpettersen) [inline sample/src/main/res/values/strings.xml:11]: nitpicky tuesday morning: "next screen" is not quite accurate for the popup that comes

File: sample/src/main/kotlin/com/pexip/sdk/sample/permissions/PermissionsAction.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+typealias PermissionsAction = WorkflowAction<PermissionsProps, PermissionsState, PermissionsOutput>
+
+class OnPermissionsRequestResult(private val result: PermissionsRequestResult) :
+    PermissionsAction() {
+
+    override fun Updater.apply() {
+        val output = when {
+            result.grants.all() -> PermissionsOutput.Next
+            result.rationales.none() && state.rationales.none() -> PermissionsOutput.ApplicationDetailsSettings
```
Comment (tpettersen) [inline sample/src/main/kotlin/com/pexip/sdk/sample/permissions/PermissionsAction.kt:13]: This triggers a slightly weird behavior. When seeing the dialog for the first time, pressing outside the dialog, I get sent to ApplicationDetails. I would expect the dialog to just close. That does not happen if I at some point remove 1 setting, gets the popup again and press outside the popup

**Review by tpettersen — APPROVED** [review]
It works good and feels better with an explanation up front 👍  I had the tiniest nitpicks that you can do whatever you want with.

File: sample/src/main/res/values/strings.xml (hunk 1 above)
Comment (drymarau) [reply sample/src/main/res/values/strings.xml:11]: Copied from ascari, and more generic (nothing stopping OEMs from making it an actual "screen" too)

File: sample/src/main/kotlin/com/pexip/sdk/sample/permissions/PermissionsAction.kt (hunk 2 above)
Comment (tpettersen) [reply sample/src/main/kotlin/com/pexip/sdk/sample/permissions/PermissionsAction.kt:13]: Does not sound like something worth fixing 😅 Super minor case

**Pull request merged**